docs/reference/api-v1alpha.md

```diff
@@ -916,4 +916,540 @@ curl -i \
+     -H "Content-Type: application/json"
+```
+
+### Deleting a target
+
+This API endpoint allows you to delete a specific deployment target.
+
+```
+DELETE {org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id?unique_token=:unique_token
+```
+
+**Params**
+
+- `target_id` (**required**) - The UUID of the deployment target to be deleted.
+- `unique_token` (**required**) - The idempotency UUID token.
+
+**Response**
+
+```json
+HTTP status: 200
+
+{"target_id":"38572f07-15ec-459e-a122-eefa2bd19230"}
+```
+
+**Example**
+
+```
+curl -i -X DELETE -H "Authorization: Token {api_token}" \
+     "https://{org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id?unique_token=:unique_token"
+```
+
+### Deactivating a target
+
+This API endpoint allows you to deactivate a specific deployment target.
+
+```
+PATCH {org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id/deactivate
+```
+
+**Params**
+
+- `target_id` (**required**) - The UUID of the deployment target to be deactivated.
+
+**Response**
+
+```json
+HTTP status: 200
+
+{
+  "target_id": "a3db05c5-4345-493d-a038-6b5f2f2ba2b5",
+  "cordoned": true
+}
+```
+
+**Example**
+
+```
+curl -i -X PATCH -H "Authorization: Token {api_token}" \
+     "https://{org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id/deactivate"
+```
+
+
+### Activating a target
+
+This API endpoint allows you to reactivate a specific deployment target.
+
+```
+PATCH {org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id/activate
+```
+
+- `target_id` (**required**) - The UUID of the deployment target to be activated.
+
+**Response**
+
+```
+HTTP status: 200
+
+{
+  "target_id": "a3db05c5-4345-493d-a038-6b5f2f2ba2b5",
+  "cordoned": false
+}
+```
+
+**Example**
+
+```
+curl -i -X PATCH -H "Authorization: Token {api_token}" \
+     "https://{org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id/activate"
+```
+
+### Retrieving deployment history
+
+This endpoint provides the deployment history for a specific deployment target.
+
+```
+GET {org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id/history
+```
+
+**Params**
+
+- `target_id` (**required**) - The UUID of the deployment target for which the deployment history is being retrieved.
+- `cursor_type` (*optional*) - Defines the starting point for data retrieval. Possible values: `"FIRST"`, `"AFTER"`, `"BEFORE"`.
+- `cursor_value` (*optional*) - Specific value for the data retrieval cursor to refer to. Works in conjunction with `cursor_type`.
+- `filters` (*optional*) - Filters that you may apply to narrow down the returned history results.
```

Comment:
We should list all available filters here and add short explanation for each of them.